packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -212,3 +216,175 @@ class _TweenAnimationBuilderState<T extends Object?>
+    this.reverse = false,
+    this.paused = false,
+    required this.builder,
+    this.child,
+  });
+
+  /// The animatable to drive repeatedly.
+  ///
+  /// The animation continuously cycles from the value returned by evaluating
+  /// [animatable] at 0.0 to the value returned at 1.0.
+  /// In most cases this will be a [Tween] or [TweenSequence], but any
+  /// [Animatable] that produces values of type [T] is accepted.
+  ///
+  /// ## Ownership
+  ///
+  /// The [RepeatingTweenAnimationBuilder] takes ownership of the provided
+  /// [Animatable] and may mutate it. Once an [Animatable] instance has been
+  /// passed to [RepeatingTweenAnimationBuilder] its properties should not be
+  /// accessed or changed to avoid interference with the
+  /// [RepeatingTweenAnimationBuilder].
+  ///
+  /// It is good practice to never store an [Animatable] provided to a
+  /// [RepeatingTweenAnimationBuilder] in an instance variable to avoid accidental
+  /// modifications of the [Animatable].
+  final Animatable<T> animatable;
+
+  /// The duration of the animation.
+  final Duration duration;
+
+  /// The curve to apply when animating the value.
+  ///
+  /// Defaults to [Curves.linear].
+  final Curve curve;
+
+  /// Called every time the animation value changes.
+  ///
+  /// The builder receives the current animation value and the optional child.
+  final ValueWidgetBuilder<T> builder;
+
+  /// The child widget to pass to the builder.
+  ///
+  /// If a builder callback's return value contains a subtree that does not
+  /// depend on the animation, it's more efficient to build that subtree once
+  /// instead of rebuilding it on every animation tick.
+  final Widget? child;
+
+  /// Whether the animation should reverse direction when it reaches the end.
+  ///
+  /// When true, the animation reverses direction instead of jumping back to
+  /// the beginning. Defaults to false.
+  final bool reverse;
+
+  /// Whether the animation is currently paused.
+  ///
+  /// When true, the animation stops at its current value. When changed to
+  /// false, the animation resumes from that value. Defaults to false.
+  final bool paused;
+
+  @override
+  State<RepeatingTweenAnimationBuilder<T>> createState() {
+    return _RepeatingTweenAnimationBuilderState<T>();
+  }
+}
+
+class _RepeatingTweenAnimationBuilderState<T extends Object>
+    extends State<RepeatingTweenAnimationBuilder<T>>
+    with SingleTickerProviderStateMixin {
+  late final AnimationController _controller;
+  late Animation<T> _animation;
+  CurvedAnimation? _curvedAnimation;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(duration: widget.duration, vsync: this);
+    final T startValue = widget.animatable.transform(0.0);
+    final T endValue = widget.animatable.transform(1.0);
+    _updateAnimation();
+
+    // Start the animation if not paused and the animation spans different values.
+    if (!widget.paused && startValue != endValue) {
```

Comment:
(GitHub seems to have lost my comment?)

Now that this accepts an `Animatable`, I can pass a `TweenSequence<Offset>` that goes right, then goes left back to where it started. We should run the animation even if the start and end values are the same.